pkg/bus/bus.go

```diff
@@ -143,9 +143,9 @@ func (b *InProcBus) Publish(msg Msg) error {
 
 	for _, listenerHandler := range listeners {
 		ret := reflect.ValueOf(listenerHandler).Call(params)
-		err := ret[0].Interface()
-		if err != nil {
-			return err.(error)
+		err, ok := ret[0].Interface().(error)
+		if ok && err != nil {
+			return err
 		}
```

Comment:
Thought, maybe it's a good thing to safe up by making sure that if it's _not_ an error we complain

```suggestion
		if ok && err != nil {
			return err
		} else if !ok {
			return fmt.Errorf("expected listener to return an error, got '%v'", reflect.TypeOf(ret[0].Interface()))
		}
```